client/my-sites/stats/activity-log-banner/progress-banner.jsx

```diff
@@ -44,48 +44,61 @@ function ProgressBanner( {
 	restoreId,
 	downloadId,
 	action,
+	context,
 } ) {
+	let title = '';
+	let description = '';
+	let statusMessage = '';
+
+	if ( 'restore' === action && 'alternate' === context ) {
+		title = translate( 'Currently cloning your site' );
+		description = translate(
+			"We're in the process of cloning your site to %s. " +
+				"You'll be notified once it's complete.",
+			{ args: applySiteOffset( moment.utc( ms( timestamp ) ) ).format( 'LLLL' ) }
+		);
+		statusMessage =
+			'queued' === status
+				? translate( 'The cloning process will start in a moment.' )
+				: translate( "We're on it! Your site is being cloned." );
+	} else if ( 'restore' === action ) {
+		title = translate( 'Currently restoring your site' );
+		description = translate(
+			"We're in the process of restoring your site back to %s. " +
+				"You'll be notified once it's complete.",
+			{ args: applySiteOffset( moment.utc( ms( timestamp ) ) ).format( 'LLLL' ) }
+		);
+		statusMessage =
+			'queued' === status
+				? translate( 'Your restore will start in a moment.' )
+				: translate( "We're on it! Your site is being restored." );
+	} else {
```

Comment:
I think these if-else if-else are a bit complicated to read. Why not group them by action, restore or not, and if it's a restore, by context? Something like:
```js
if ( 'restore' === action ) {
    if ( 'alternate' === context ) {
        // clone
    } else {
        // restore
    }
} else {
    // backup
}
```